src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,63 @@
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  filter: string;
+  query: string;
+  onFilterChange: (status: string) => void;
+  onSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onClearSearch: () => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  filter,
+  query,
+  onFilterChange,
+  onSearchChange,
+  onClearSearch,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={filter}
+            onChange={e => onFilterChange(e.target.value)}
+            data-cy="statusSelect"
+            aria-label="Filter tasks by status"
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
```

Comment:
Consider using the `classnames` library for managing conditional classes, as specified in the task requirements. This will help improve code readability and maintainability.